src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkConnectorConfig.java

```diff
@@ -495,8 +513,14 @@ public boolean dropInvalidMessage() {
         return getBoolean(OpensearchSinkConnectorConfig.DROP_INVALID_MESSAGE_CONFIG);
     }
 
-    public boolean ignoreKeyFor(final String topic) {
-        return ignoreKey() || topicIgnoreKey().contains(topic);
+    public RecordConverter.DocumentIDStrategy docIdStrategy() {
+        // Default is TOPIC_PARTITION_OFFSET if unspecified
+        return RecordConverter.DocumentIDStrategy.fromString(getString(KEY_IGNORE_ID_STRATEGY_CONFIG));
+    }
+
+    public RecordConverter.DocumentIDStrategy globalDocIdStrategy() {
```

Comment:
I believe we don't need both methods, it is confusing which one does what, having just one should be sufficient:
```
public RecordConverter.DocumentIDStrategy docIdStrategy(String topic) {
    return `ignoreKey() ||  topicIgnoreKey().contains(topic)` ? RecordConverter.DocumentIDStrategy.fromString(getString(KEY_IGNORE_ID_STRATEGY_CONFIG) : RecordConverter.DocumentIDStrategy.RECORD_KEY;
}
```

Reply:
(to mitigate possible impact on re-checking configuration every time, we could may be cache it inside `RecordConverter`)